### PR TITLE
First stab at QueryDSL

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: quickwit/quickwit-serve/src/elastic_search_api/rest_handler.rs [hunk 1]
```diff
@@ -38,45 +48,75 @@ pub fn elastic_get_search_handler(
...
+    indexes: SimpleList,
+    search_params: SearchQueryParams,
+    search_service: Arc<dyn SearchService>,
+) -> Result<ElasticSearchResponse, SearchError> {
+    let start_instant = Instant::now();
+    if indexes.0.len() != 1 {
+        let error_msg = format!("Expected exaclty one index, got {indexes:?}.");
+        return Err(SearchError::InvalidArgument(error_msg));
+    }
+    let index_id: String = indexes.0.into_iter().next().unwrap();
```
Comment (guilload) [inline quickwit/quickwit-serve/src/elastic_search_api/rest_handler.rs:71]:
```suggestion
    let index_id: String = indexes.0.into_iter().next().expect("There should be exactly once index in the list, as checked by the if statement above.");
```

File: quickwit/quickwit-serve/src/elastic_search_api/rest_handler.rs [hunk 2]
```diff
@@ -38,45 +48,75 @@ pub fn elastic_get_search_handler(
...
+        hits: HitsMetadata {
+            total: Some(TotalHits {
+                value: resp.num_hits,
+                relation: TotalHitsRelation::Equal,
+            }),
+            max_score: None,
+            hits,
         },
-    )
+        // terminated_early: todo!(),
```
Comment (guilload) [inline quickwit/quickwit-serve/src/elastic_search_api/rest_handler.rs:118]: Remove comment or implement or open an issue?

**Review by guilload — APPROVED** [review]

File: quickwit/Cargo.toml [hunk 3]
```diff
@@ -204,7 +204,7 @@ quickwit-control-plane = { version = "0.5.0", path = "./quickwit-control-plane"
 quickwit-core = { version = "0.5.0", path = "./quickwit-core" }
 quickwit-directories = { version = "0.5.0", path = "./quickwit-directories" }
 quickwit-doc-mapper = { version = "0.5.0", path = "./quickwit-doc-mapper" }
-quickwit_elastic_api_generation = { git = "https://github.com/quickwit-oss/elasticsearch-rs", rev = "dca3aba" }
+quickwit_elastic_api_generation = { git = "https://github.com/quickwit-oss/elasticsearch-rs", rev = "7c57830" }
```
[the remark this answers is not in the document]
Comment (fulmicoton) [reply quickwit/Cargo.toml:207]: done

File: quickwit/quickwit-search/src/lib.rs [hunk 4]
```diff
@@ -30,6 +30,7 @@ mod fetch_docs;
 mod filters;
 mod find_trace_ids_collector;
 mod leaf;
+mod query_dsl;
```
[the remark this answers is not in the document]
Comment (fulmicoton) [reply quickwit/quickwit-search/src/lib.rs:33]: in the future :)

**Pull request merged**